### PR TITLE
Feature/fix refactor backlog service

### PR DESCRIPTION
**Pull request by chevyB**
Issue: https://app.asana.com/0/1202107027507930/1202266969657290/f

Definition:
- refactor the existing test in backlog service (all test)
- should not persist in the actual database (do mocking)
- apply proper test names (followed in standards)

Files Affected:
- backlog/test/controllers/ProjectController.test.ts
- backlog/test/controllers/ProviderController.test.ts
- backlog/test/routes/ProjectRoute.test.ts _(deleted)_

### Commands to run
- `docker-compose exec backlog yarn test`

### Services to run
- db
- backlog

### Screenshot
image

### CODE REVIEW

**Comment by NielmarLaurente** [comment]
### Comment  : 
- I have these error in the `network` and `console` tab .  I tried to add new project and then go back to home project list Page.

image

image

image

image

image

File: backlog/test/controllers/BacklogController.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    req = httpMocks.createRequest()
+    res = httpMocks.createResponse()
+    Controller = new BacklogController()
+  })
+
+  it('should return 404 error if apikey is wrong', async () => {
+    req.query = payload
+    await Controller.getList(req, res)
+
+    console.log({ test: res._getData() })
```
Comment (escbooster12-dev) [inline backlog/test/controllers/BacklogController.test.ts:32]: - there are console logs in the terminal. is it okay to remove it? :man: 
image

File: backlog/test/controllers/ProviderController.test.ts [hunk 2]
```diff
@@ -2,78 +2,113 @@ require('dotenv').config()
...
 import { Request, Response } from 'express'
-import { faker } from '@faker-js/faker'
-const Provider = require('../../models/Provider')
+import Provider from '../../models/Provider'
+import { prismaMock } from '../../utils/singleton'
 const axios = require('axios')
 const MockAdapter = require('axios-mock-adapter')
 
+const mockedProviderResponse = {
+  id: 1,
```
Comment (escbooster12-dev) [inline backlog/test/controllers/ProviderController.test.ts:11]: - please move constant variables to `test/const` directory :bow:

**Review by NielmarLaurente — APPROVED** [review]
## Approved PR #30 
### Working :
- Test Passed for the ff

> backlog/test/controllers/ProjectController.test.ts
> backlog/test/controllers/ProviderController.test.ts
> backlog/test/routes/ProjectRoute.test.ts (deleted)
- No error in the console and network tab
- Functionalities are also working well.

<details>

<summary> Screenshot </summary>

image

https://user-images.githubusercontent.com/58454589/169759865-7dd17f5b-7bc7-4fd4-87e8-b39c4d0311a2.mp4



</details>

**Testing Approval time : 0.75 hr
Total comments : 1
Total Iterations : 1**

File: backlog/test/controllers/BacklogController.test.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import { Request, Response } from 'express'
+const axios = require('axios')
+const MockAdapter = require('axios-mock-adapter')
+
+describe('When getting project list from external backlog api', () => {
+  interface TypedResponse extends Response {
+    statusCode: any
+    _getData: () => any
+  }
+
```
Comment (escbooster12-dev) [inline backlog/test/controllers/BacklogController.test.ts:14]: - can we make this one as reusable interface. i think this one was also repeatedly declared in file **backlog/test/controllers/ProjectController.test.ts**

**Review by escbooster12-dev — APPROVED** [review]
- tests are working

**Pull request merged**